ra/ra.go

```diff
@@ -2325,17 +2430,30 @@ func (ra *RegistrationAuthorityImpl) NewOrder(ctx context.Context, req *rapb.New
 	// authorization for each.
 	var newAuthzs []*sapb.NewAuthzRequest
 	for _, ident := range missingAuthzIdents {
-		pb, err := ra.createPendingAuthz(newOrder.RegistrationID, ident)
+		challTypes, err := ra.PA.ChallengeTypesFor(ident)
 		if err != nil {
 			return nil, err
 		}
-		newAuthzs = append(newAuthzs, pb)
+
+		challStrs := make([]string, len(challTypes))
+		for i, t := range challTypes {
+			challStrs[i] = string(t)
+		}
```

Comment:
This is moved code, so feel free to fix this as a followup, but I think we've decided against preallocating slices in non-performance-critical settings, and particularly against this form where it's possible to observe preallocated zero values by accident.

```suggestion
		var challStrs []string
		for _, t := range challTypes {
			challStrs = append(challStr, string(t))
		}
```